### PR TITLE
switch to signup page from login page

### PR DESCRIPTION
**Pull request by alexrosen45**
Switch to signup page from login page using a new `switch_view` use case interactor. One potential issue is that after a user makes an account, if they click the switch to signup button on the login page, they will be presented with the information they already filled in. This is probably good for when we switch from the signup page to the login page if the user was on the login page previously, but we should get rid of it in the login page to signup page case. I'm holding off on it now because it's better to implement this after we have a signup page to login page button.

### CODE REVIEW

File: src/view/LoginView.java [hunk 1]
```diff
@@ -14,104 +14,109 @@
...
+
+        LabelTextPanel usernameInfo = new LabelTextPanel(new JLabel("Username"), usernameInputField);
+        LabelTextPanel passwordInfo = new LabelTextPanel(new JLabel("Password"), passwordInputField);
+
+        JPanel buttons = new JPanel();
+        logIn = new JButton(LoginViewModel.LOGIN_BUTTON_LABEL);
+        buttons.add(logIn);
+        cancel = new JButton(LoginViewModel.CANCEL_BUTTON_LABEL);
+        buttons.add(cancel);
+        switchToSignup = new JButton("Sign Up");
```
Comment (alexrosen45) [inline src/view/LoginView.java:48]: We should also change this to a `LoginViewModel.SWITCH_TO_SIGNUP_BUTTON_LABEL` soon.

**Comment by uoftm** [comment]
Looks like formatting is causing problems here... Can you try running `mvn com.spotify.fmt:fmt-maven-plugin:format`?

File: src/interface_adapter/login/LoginController.java [hunk 2]
```diff
@@ -2,18 +2,26 @@
...
   }
 
   public void execute(String username, String password) {
     LoginInputData loginInputData = new LoginInputData(username, password);
 
     loginUseCaseInteractor.execute(loginInputData);
   }
+
+  public void switchToSignup() {
+    switchViewInteractor.execute(new SwitchViewInputData("sign up"));
```
Comment (uoftm) [inline src/interface_adapter/login/LoginController.java:25]: Magic string, see my question on using statics as a different way to do this https://github.com/uoftm/csc207project/pull/6

File: src/view/LoginView.java [hunk 3]
```diff
@@ -59,6 +62,8 @@ public void actionPerformed(ActionEvent evt) {
 
     cancel.addActionListener(this);
 
+    switchToSignup.addActionListener(e -> loginController.switchToSignup());
```
Comment (uoftm) [inline src/view/LoginView.java:65]: I like this lambda :)

File: src/use_case/switch_view/SwitchViewUserDataAccessInterface.java [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package use_case.switch_view;
+
+public interface SwitchViewUserDataAccessInterface {
```
Comment (uoftm) [inline src/use_case/switch_view/SwitchViewUserDataAccessInterface.java:3]: Delete this file?

Comment (alexrosen45) [reply src/use_case/switch_view/SwitchViewUserDataAccessInterface.java:3]: Sure, yeah. Wasn't sure whether to leave it in or not.

File: src/interface_adapter/login/LoginController.java [hunk 5]
```diff
@@ -2,18 +2,27 @@
...
 public class LoginController {
 
   final LoginInputBoundary loginUseCaseInteractor;
+  private final SwitchViewInputBoundary switchViewInteractor;
 
-  public LoginController(LoginInputBoundary loginUseCaseInteractor) {
+  public LoginController(
+      LoginInputBoundary loginUseCaseInteractor, SwitchViewInputBoundary switchViewInteractor) {
     this.loginUseCaseInteractor = loginUseCaseInteractor;
+    this.switchViewInteractor = switchViewInteractor;
```
Comment (uoftm) [inline src/interface_adapter/login/LoginController.java:16]: Aren't we expected to have a SwitchViewController?

Comment (alexrosen45) [reply src/interface_adapter/login/LoginController.java:16]: Not for a use case

File: src/interface_adapter/login/LoginController.java (hunk 2 above)
Comment (alexrosen45) [reply src/interface_adapter/login/LoginController.java:26]: Let's meet as a team to discuss this. I think it's ok for now

File: src/view/LoginView.java [hunk 6]
```diff
@@ -59,6 +68,16 @@ public void actionPerformed(ActionEvent evt) {
...
 
+    switchToSignup.addActionListener(
+        new ActionListener() {
+          @Override
+          public void actionPerformed(ActionEvent e) {
+            switchViewController.switchTo("sign up");
+          }
+        });
+
+    cancel.addActionListener(this);
```
Comment (modderme123) [inline src/view/LoginView.java:79]: ?

File: src/view/LoginView.java [hunk 7]
```diff
@@ -59,6 +68,16 @@ public void actionPerformed(ActionEvent evt) {
 
     cancel.addActionListener(this);
 
+    switchToSignup.addActionListener(
+        new ActionListener() {
+          @Override
+          public void actionPerformed(ActionEvent e) {
+            switchViewController.switchTo("sign up");
```
Comment (modderme123) [inline src/view/LoginView.java:75]: I don't like this string, but we'll fix it once we've merged #6

File: src/view/LoginView.java (hunk 6 above)
Comment (alexrosen45) [reply src/view/LoginView.java]: fixed

File: src/view/LoginView.java (hunk 7 above)
Comment (alexrosen45) [reply src/view/LoginView.java:75]: sure

File: src/interface_adapter/login/LoginPresenter.java [hunk 8]
```diff
@@ -5,8 +5,10 @@
 import interface_adapter.logged_in.LoggedInViewModel;
 import use_case.login.LoginOutputBoundary;
 import use_case.login.LoginOutputData;
+import use_case.switch_view.SwitchViewOutputBoundary;
+import use_case.switch_view.SwitchViewOutputData;
 
-public class LoginPresenter implements LoginOutputBoundary {
+public class LoginPresenter implements LoginOutputBoundary, SwitchViewOutputBoundary {
```
Comment (uoftm) [inline src/interface_adapter/login/LoginPresenter.java:11]: We definitely want a SwitchViewPresenter

Comment (alexrosen45) [reply src/interface_adapter/login/LoginPresenter.java:11]: Let's add this as a second PR

**Review by uoftm — APPROVED** [review]
Good enough to merge

**Pull request merged**